### DISCUSSION TITLE
Plot the Drug amount into Stratum Corneum

### DISCUSSION THREAD

**Discussion by jonathanlsn** [Open-Systems-Pharmacology/Forum#871] [Questions & Problems] [answered]
Hi everyone !
I'm just started with OSP M
<img width="761" alt="Mobi_screenshot" src="https://user-images.githubusercontent.com/47557630/109007450-68d25780-76ac-11eb-9ed4-c4d95675c83f.png">
obi and PKsim and I got some issues I will try to explain.

I just made a model with a skin compartment which is linked to the whole body system on Mobi. I wanted to simulate the permeation of a drug through the different layers of the skin. I had selected Plasma, SC_skin (Stratum Cormeum) and DE_skin (Dermis) drug amount into simulation settings but when I launched the simulation I only see the amount of drug into the Plasmatic compartment. So there are no curves for the amount of drug into the Stratum Corneum and the Dermis even if I select them one by one.

If you have any solution please don't hesitate to reply.
Thank you in advance.

**Comment by abdullahhamadeh** [comment]
Hi @jonathanlsn 

Did you set a non-zero dose (in mass per area) in DERMAL_APPLICATION_AREA|skin_compartment|Dose?

What happens when you select to plot vehicle_mass in Simulation Settings?

**Comment by jonathanlsn** [comment]
I put 2000 mg/m2 for the Dose value. 
When I select the vehicle_mass in the Simulation Settings I don't see any plot neither. But I still got the curve for the plasmatic compartment.

**Comment by abdullahhamadeh** [comment]
And if you double or halve this dose in DERMAL_APPLICATION_AREA|skin_compartment|Dose do you see a corresponding change in plasma?  

Do you have an application that was added in PKSim that is still in the linked model? It may account for the drug that is entering plasma.

**Comment by jonathanlsn** [comment]
I double the dose and run a new simulation with this value but it doesn't change anything to the amount of drug into the plasma.

I made an administration protocol in PKsim and I it is still linked to the model because I merged them. For this protocol I selected a User define application, target organ : skin, target compartment : plasma. We can see the affect of the administration protocol on the curve in blue, linked to the plasma compartment.

<img width="693" alt="Mobi_screenshot_plasma" src="https://user-images.githubusercontent.com/47557630/109029110-cc677f80-76c2-11eb-818f-844aaf9b43f1.png">

Update : now I see the plot for the amount of drug into the SC_skin compartment after changed the scalling to linear mode. But the value stay to 0.

**Comment by jonathanlsn** [comment]
Finally I succeed to plot a non-null curve for SC, DE and ED_skin after using an older version of the skin_permeation_model I found on your Github. Perhaps the latest version doesn't work as I wanted.

I also got a other issue, maybe linked to the previous one. When I try to merge the skin_simulation I made on Mobi and the whole_body_simulation with administration protocol I made on PKsim there is a conflict between two molecules named "permeant". This is the name I chose for my molecule. The first one got 90 parameters and the other one got 1 parameters. I always select the one with 90 parameters and I wondered where does the second permeant come from ? I tried to run a simulation with the second one already but Mobi return an error message.
Do you know if the problem could be linked to the graphical issue ?

**Comment by abdullahhamadeh** [comment]
@jonathanlsn yes, correct to choose the 90 parameter one, which comes from the pksim model and has molecule parameters that are needed in the whole body part of the combined model.  

I can take a look at the issue with the later model version if you can share the pkml file here.

**Comment by jonathanlsn** [comment]
I share you an entire file with the skin_model, the whole_body_model and the merged_model which is the one containing the final plot. I wandered why the amount of drug stay at zero for each layers of the skin ? And this happen only with the latest version of skin_permeation_model.

[Octanol model.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/6062802/Octanol.model.zip)

**Comment by jonathanlsn** [comment]
Hello,
Did you watch the model so you can understand what is wrong with this version of skin_model ? 
Please tell me if you have any idea about where this issue come from.

**Comment by abdullahhamadeh** [comment, marked as answer]
@jonathanlsn 

There are no dermal applications in your merged whole body model.  To add them, please use these steps:
1 - download the [events building block from here](https://raw.githubusercontent.com/Open-Systems-Pharmacology/Skin-permeation-model/master/skin_events_remove_then_add.pkml)

2 - in Octanol_merged_model_project.mbp3 go to the 'Events' building block, double click 'Octanol_whole_body_simulation', right click on 'Applications', and select 'Load Application', as shown here: 
![image](https://user-images.githubusercontent.com/18409098/115973947-cfe28000-a526-11eb-931a-18ddb825982e.png)

3 - select the file you downloaded in step 1 above.  This will load the application called  'skin_application'.  There are two events you can modify or delete:

![image](https://user-images.githubusercontent.com/18409098/115973970-2e0f6300-a527-11eb-8407-3fdc8cf98279.png)

4 - Delete 'event_with_dose_mass_entered_manually' and keep 'event_with_dose_mass_from_spatial_structure'.  Now, you can set the dermal dose from the favorites menu or by modifying it directly from DERMAL_APPLICATION_AREA|skin_compartment|Dose.

5 - Update the 'Events' building block in the simulation called 'Octanol_merged_model_simulation' by clicking on the simulation, then on 'Configuration', then and then right-clicking on the events group for the simulation and selecting 'Update from building block', as shown here:
![image](https://user-images.githubusercontent.com/18409098/115974093-521f7400-a528-11eb-8e95-21f6f54fa24e.png)

6 - Re-run the simulation.  Now you will be able to see the contribution from the dermal application:
![image](https://user-images.githubusercontent.com/18409098/115974194-189b3880-a529-11eb-95a0-8cdc133bc7bd.png)

**Comment by jonathanlsn** [comment]
@abdullahhamadeh 
Thank you for these explanations, I can see all curves now. But I have an other question concerning the drug clearance into venous blood. When I expand the duration of the simulation from 112h to 300h I was expecting to observe a decreasing of the amount of drug to 0 but this amount reach non nul fixed state because of its degradation into the body. Do you know where does this problem come from ?

![image](https://user-images.githubusercontent.com/47557630/117288717-1495e180-ae6c-11eb-8691-ebad0651b544.png)

**Reply by abdullahhamadeh** [reply]
Hi @jonathanlsn  

I think you might have some of the permeant left in the vehicle compartment and you have continuous flux into the plasma from the dermal compartment.  To check, plot the vehicle compartment amount and also the flux into plasma, 'J'.

**Reply by jonathanlsn** [reply]
Hi @abdullahhamadeh 

As you thought there is a continuous flux from dermal compartment to plasma and even 300 hours after the first administration there is a high amount of permeant left in vehicle compartment. I was expected the permeant amount in vehicle would be lower than what I observed in this simulation. Is there a way to modify the flux from vehicule to skin compartment ?

![image](https://user-images.githubusercontent.com/47557630/118643282-a0006280-b7dc-11eb-9a78-3c930c2f7e1a.png)
